echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/PipelineCache.java

```diff
@@ -24,70 +25,98 @@
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
-import rx.Observable;
-import rx.Scheduler;
-import rx.Subscription;
-import rx.subjects.ReplaySubject;
-import rx.subjects.SerializedSubject;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import javax.annotation.PostConstruct;
 import javax.annotation.PreDestroy;
 import java.time.Instant;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 import java.util.stream.Collectors;
 
 import static java.time.Instant.now;
-import static java.util.concurrent.TimeUnit.SECONDS;
 
 @Component
 @Slf4j
 public class PipelineCache implements MonitoredPoller {
-  private final Scheduler scheduler;
   private final int pollingIntervalSeconds;
   private final Front50Service front50;
   private final Registry registry;
+  private final ScheduledExecutorService executorService;
 
+  private transient Boolean running;
   private transient Instant lastPollTimestamp;
-  private transient Subscription subscription;
 
-  private transient SerializedSubject<List<Pipeline>, List<Pipeline>> pipelineSubject = ReplaySubject.<List<Pipeline>>createWithSize(1).toSerialized();
+  @Nullable
+  private transient List<Pipeline> pipelines;
 
   @Autowired
-  public PipelineCache(@NonNull Scheduler scheduler,
+  public PipelineCache(int pollingIntervalSeconds,
+                       @NonNull Front50Service front50,
+                       @NonNull Registry registry) {
+    this(Executors.newSingleThreadScheduledExecutor(), pollingIntervalSeconds, front50, registry);
+  }
+
+  // VisibleForTesting
+  public PipelineCache(ScheduledExecutorService executorService,
                        int pollingIntervalSeconds,
                        @NonNull Front50Service front50,
                        @NonNull Registry registry) {
-    this.scheduler = scheduler;
+    this.executorService = executorService;
     this.pollingIntervalSeconds = pollingIntervalSeconds;
     this.front50 = front50;
     this.registry = registry;
+    this.running = false;
+    this.pipelines = null;
+  }
+
+  @PreDestroy
+  public void stop() {
+    running = false;
```

Comment:
Shouldn't we also be doing `executorService.shutdown()` here as well?

Reply:
we probably should

Reply:
I think we previously had tests that asserted that we could `stop()` then `start()` again. But given my understanding of `@PreDestroy`, this is more of a one-way street so probably safe to shutdown the executor service here